### PR TITLE
hasmail: init at 2019-08-24

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/hasmail/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}:
+
+buildGoModule rec {
+  pname = "hasmail";
+  version = "2019-08-24";
+  name = "${pname}-${version}-unstable";
+
+  src = fetchFromGitHub {
+    owner = "jonhoo";
+    repo = "hasmail";
```
Comment (mmahut) [inline pkgs/applications/networking/mailreaders/hasmail/default.nix]:
```suggestion
    repo = pname;
```

Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: I've had reviews in the past which convinced me to not use `pname` in the `fetchFromGitHub` function. I don't have such a strong opinion about this but I'd like to be at least consistent in all of my _own_ PRs.

Comment (mmahut) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: Interesting, do you have any reference?

Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: https://github.com/NixOS/nixpkgs/pull/65480#discussion_r315799884

File: pkgs/applications/networking/mailreaders/hasmail/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, pkgconfig
+, gobject-introspection
+, pango
+, cairo
+, gtk2
```
Comment (aanderse) [inline pkgs/applications/networking/mailreaders/hasmail/default.nix]: The program doesn't support `gtk3`?

File: pkgs/applications/networking/mailreaders/hasmail/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}:
+
+buildGoModule rec {
+  pname = "hasmail";
+  version = "2019-08-24";
+  name = "${pname}-${version}-unstable";
```
Comment (aanderse) [inline pkgs/applications/networking/mailreaders/hasmail/default.nix]: I thought we were adding `-unstable` to `pname` now...

File: pkgs/applications/networking/mailreaders/hasmail/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    owner = "jonhoo";
+    repo = "hasmail";
+    rev = "eb52536d26815383bfe5990cd5ace8bb9d036c8d";
+    sha256 = "1p6kwa5xk1mb1fkkxz1b5rcyp5kb4zc8nfif1gk6fab6wbdj9ia1";
+  };
+
+  modSha256 = "0z3asz7v1izg81f9xifx9s2sp5hly173hajsn9idi3bkv0h78is2";
+
+  buildInputs = [
+    pkgconfig
```
Comment (aanderse) [inline pkgs/applications/networking/mailreaders/hasmail/default.nix]: This belongs in `nativeBuildInputs`.

File: pkgs/applications/networking/mailreaders/hasmail/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    gobject-introspection
```
Comment (aanderse) [inline pkgs/applications/networking/mailreaders/hasmail/default.nix]: Do you need `wrapGAppsHook`? I've never looked at this program so I can't say. See https://github.com/NixOS/nixpkgs/pull/43150/files for documentation if required.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/mailreaders/hasmail/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: I get the error:

```
# pkg-config --cflags  -- gdk-2.0 gthread-2.0
Package gdk-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gdk-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gdk-2.0' found
pkg-config: exit status 1
```

And according to `nix-locate gdk-2.0.pc`, only GTK 2 provides this file.

File: pkgs/applications/networking/mailreaders/hasmail/default.nix (hunk 3 above)
Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: According to e.g `xsel`, you are right.

File: pkgs/applications/networking/mailreaders/hasmail/default.nix (hunk 4 above)
Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: :+1:

File: pkgs/applications/networking/mailreaders/hasmail/default.nix (hunk 5 above)
Comment (doronbehar) [reply pkgs/applications/networking/mailreaders/hasmail/default.nix]: No, you were right. I wonder why I thought I'd need it in the first place..

**Review by doronbehar — COMMENTED** [review]
Thanks for all the comments @aanderse I think I've fixed all of the issues.

**Comment by doronbehar** [comment]
Thanks @matthewbauer! Could you check please https://github.com/NixOS/nixpkgs/pull/69843?

**Pull request merged**